tests/unit/specification/grammar_test.py

```diff
@@ -816,7 +822,7 @@ def test_state_error(string: str, error: str) -> None:
                generic
                   X : Channel with Readable, Writable;
                   type T is private;
-                  with function F return T;
+                  with function F return Package::T;
                session Session with
                   Initial => A,
                   Final => B
```

Comment:
I'm wondering why just test cases are modified in this file, but no test cases are added. Shouldn't qualified identifiers as well as non-qualified identifiers be supported and also tested? IMHO, it would make more sense to add another test case to `test_formal_function_declaration` instead of changing the return type here.

Reply:
Added an additional test with an unqualified return type.

Reply:
Copying the complete session declaration test case of `test_session_declaration` and just changing a single identifier is not what I had in mind. Unit tests should be as small as possible. If you want to test, that (un)qualified identifier work for formal function declarations, a test case should be added to `test_formal_function_declaration`, which would would be just a single line instead around 60 lines which were added here.

Reply:
You are right, `test_formal_function_declaration` is the better place. I changed that.